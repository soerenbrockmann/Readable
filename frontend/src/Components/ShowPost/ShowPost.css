.showpost {
  display: flex;
  flex-direction: column;
  border: 1px solid #ddd;
  padding: 10px;
  margin-top: 1em;
  background-color: #f8f8f8;
}

.showpostheader {
  display: flex;
  flex-direction: row;
}

.showpostheader > div {
  margin-right: 0.5em;
}

.showpostheader > div:nth-child(2) {
  margin-left: auto;
}

.showpostnav {
  display: flex;
  flex-direction: row;
  margin-top: 1em;
}

.showpostnav > div {
  margin-right: 0.5em;
}

.showpostnav > div:nth-child(2) {
  margin-left: auto;
}
