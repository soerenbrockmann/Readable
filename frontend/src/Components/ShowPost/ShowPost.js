import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import Votes from "../Votes";
import FormatDate from "../FormatDate";
import "./ShowPost.css";

const ShowPost = props => {
  const { onDelete, onEdit, onDownVote, onUpVote } = props;
  const { item } = props;
  const {
    timestamp = Date.now(),
    title = "",
    body = "",
    author = "",
    category = "",
    voteScore = 0
  } = item;

  return (
    <div className="showpost">
      <div>
        <div className="showpostheader">
          <div className="title">{title}</div>
          <div>
            <Link to={"/"} className="ondelete" onClick={onDelete}>
              Delete
            </Link>
          </div>
          <div className="onedit" onClick={onEdit}>
            Edit
          </div>
          <div>
            <Link to={"/"} className="onclose">
              Close
            </Link>
          </div>
        </div>
        <div className="meta">
          <div>{author} - </div> <FormatDate timestamp={timestamp} />
        </div>
        <div className="body">{body}</div>
        <div className="showpostnav">
          <div>
            <Votes
              onUpVote={onUpVote}
              onDownVote={onDownVote}
              voteScore={voteScore}
            />
          </div>
          <div className="category">[{category}]</div>
        </div>
      </div>
    </div>
  );
};

Votes.propTypes = {
  onUpVote: PropTypes.func.isRequired,
  onDownVote: PropTypes.func.isRequired,
  voteScore: PropTypes.number
};

FormatDate.propTypes = {
  timestamp: PropTypes.number.isRequired
};

export default ShowPost;
