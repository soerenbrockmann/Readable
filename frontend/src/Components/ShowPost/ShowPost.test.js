import React from "react";
import { shallow } from "enzyme";
import ShowPost from "./ShowPost";

describe("<ShowPost />", () => {
  const onDelete = jest.fn();
  const onEdit = jest.fn();
  const onDownVote = jest.fn();
  const onUpVote = jest.fn();
  const item = {
    timestamp: 1468166872634,
    title: "",
    body: "",
    author: "",
    category: "",
    voteScore: 0
  };
  const showPost = shallow(
    <ShowPost
      onDelete={onDelete}
      onEdit={onEdit}
      onDownVote={onDownVote}
      onUpVote={onUpVote}
      item={item}
    />
  );
  it("renders properly", () => {
    expect(showPost).toMatchSnapshot();
  });
});
