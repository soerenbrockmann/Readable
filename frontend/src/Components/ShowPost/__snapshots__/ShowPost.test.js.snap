// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowPost /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowPost
    item={
        Object {
            "author": "",
            "body": "",
            "category": "",
            "timestamp": 1468166872634,
            "title": "",
            "voteScore": 0,
          }
    }
    onDelete={[Function]}
    onDownVote={[Function]}
    onEdit={[Function]}
    onUpVote={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="showpostheader"
        >
                <div
                        className="title"
                />
                <div>
                        <Link
                                className="ondelete"
                                onClick={[Function]}
                                replace={false}
                                to="/"
                        >
                                Delete
                        </Link>
                </div>
                <div
                        className="onedit"
                        onClick={[Function]}
                >
                        Edit
                </div>
                <div>
                        <Link
                                className="onclose"
                                replace={false}
                                to="/"
                        >
                                Close
                        </Link>
                </div>
        </div>
        <div
                className="meta"
        >
                <div>
                        
                         - 
                </div>
                 
                <FormatDate
                        timestamp={1468166872634}
                />
        </div>
        <div
                className="body"
        />
        <div
                className="showpostnav"
        >
                <div>
                        <Votes
                                onDownVote={[Function]}
                                onUpVote={[Function]}
                                voteScore={0}
                        />
                </div>
                <div
                        className="category"
                >
                        [
                        
                        ]
                </div>
        </div>
</div>,
      "className": "showpost",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="showpostheader"
>
            <div
                        className="title"
            />
            <div>
                        <Link
                                    className="ondelete"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/"
                        >
                                    Delete
                        </Link>
            </div>
            <div
                        className="onedit"
                        onClick={[Function]}
            >
                        Edit
            </div>
            <div>
                        <Link
                                    className="onclose"
                                    replace={false}
                                    to="/"
                        >
                                    Close
                        </Link>
            </div>
</div>,
          <div
            className="meta"
>
            <div>
                        
                         - 
            </div>
             
            <FormatDate
                        timestamp={1468166872634}
            />
</div>,
          <div
            className="body"
/>,
          <div
            className="showpostnav"
>
            <div>
                        <Votes
                                    onDownVote={[Function]}
                                    onUpVote={[Function]}
                                    voteScore={0}
                        />
            </div>
            <div
                        className="category"
            >
                        [
                        
                        ]
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
/>,
              <div>
                <Link
                                className="ondelete"
                                onClick={[Function]}
                                replace={false}
                                to="/"
                >
                                Delete
                </Link>
</div>,
              <div
                className="onedit"
                onClick={[Function]}
>
                Edit
</div>,
              <div>
                <Link
                                className="onclose"
                                replace={false}
                                to="/"
                >
                                Close
                </Link>
</div>,
            ],
            "className": "showpostheader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "title",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="ondelete"
                  onClick={[Function]}
                  replace={false}
                  to="/"
>
                  Delete
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Delete",
                  "className": "ondelete",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Delete",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "onedit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="onclose"
                  replace={false}
                  to="/"
>
                  Close
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Close",
                  "className": "onclose",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Close",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                
                 - 
</div>,
              " ",
              <FormatDate
                timestamp={1468166872634}
/>,
            ],
            "className": "meta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  " - ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                " - ",
              ],
              "type": "div",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "timestamp": 1468166872634,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "body",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Votes
                                onDownVote={[Function]}
                                onUpVote={[Function]}
                                voteScore={0}
                />
</div>,
              <div
                className="category"
>
                [
                
                ]
</div>,
            ],
            "className": "showpostnav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Votes
                  onDownVote={[Function]}
                  onUpVote={[Function]}
                  voteScore={0}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onDownVote": [Function],
                  "onUpVote": [Function],
                  "voteScore": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "[",
                  "",
                  "]",
                ],
                "className": "category",
              },
              "ref": null,
              "rendered": Array [
                "[",
                "",
                "]",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="showpostheader"
          >
                    <div
                              className="title"
                    />
                    <div>
                              <Link
                                        className="ondelete"
                                        onClick={[Function]}
                                        replace={false}
                                        to="/"
                              >
                                        Delete
                              </Link>
                    </div>
                    <div
                              className="onedit"
                              onClick={[Function]}
                    >
                              Edit
                    </div>
                    <div>
                              <Link
                                        className="onclose"
                                        replace={false}
                                        to="/"
                              >
                                        Close
                              </Link>
                    </div>
          </div>
          <div
                    className="meta"
          >
                    <div>
                              
                               - 
                    </div>
                     
                    <FormatDate
                              timestamp={1468166872634}
                    />
          </div>
          <div
                    className="body"
          />
          <div
                    className="showpostnav"
          >
                    <div>
                              <Votes
                                        onDownVote={[Function]}
                                        onUpVote={[Function]}
                                        voteScore={0}
                              />
                    </div>
                    <div
                              className="category"
                    >
                              [
                              
                              ]
                    </div>
          </div>
</div>,
        "className": "showpost",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="showpostheader"
>
              <div
                            className="title"
              />
              <div>
                            <Link
                                          className="ondelete"
                                          onClick={[Function]}
                                          replace={false}
                                          to="/"
                            >
                                          Delete
                            </Link>
              </div>
              <div
                            className="onedit"
                            onClick={[Function]}
              >
                            Edit
              </div>
              <div>
                            <Link
                                          className="onclose"
                                          replace={false}
                                          to="/"
                            >
                                          Close
                            </Link>
              </div>
</div>,
            <div
              className="meta"
>
              <div>
                            
                             - 
              </div>
               
              <FormatDate
                            timestamp={1468166872634}
              />
</div>,
            <div
              className="body"
/>,
            <div
              className="showpostnav"
>
              <div>
                            <Votes
                                          onDownVote={[Function]}
                                          onUpVote={[Function]}
                                          voteScore={0}
                            />
              </div>
              <div
                            className="category"
              >
                            [
                            
                            ]
              </div>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
/>,
                <div>
                  <Link
                                    className="ondelete"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/"
                  >
                                    Delete
                  </Link>
</div>,
                <div
                  className="onedit"
                  onClick={[Function]}
>
                  Edit
</div>,
                <div>
                  <Link
                                    className="onclose"
                                    replace={false}
                                    to="/"
                  >
                                    Close
                  </Link>
</div>,
              ],
              "className": "showpostheader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "title",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="ondelete"
                    onClick={[Function]}
                    replace={false}
                    to="/"
>
                    Delete
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Delete",
                    "className": "ondelete",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "onedit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Edit",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="onclose"
                    replace={false}
                    to="/"
>
                    Close
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Close",
                    "className": "onclose",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Close",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  
                   - 
</div>,
                " ",
                <FormatDate
                  timestamp={1468166872634}
/>,
              ],
              "className": "meta",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "",
                    " - ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "",
                  " - ",
                ],
                "type": "div",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "timestamp": 1468166872634,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "body",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <Votes
                                    onDownVote={[Function]}
                                    onUpVote={[Function]}
                                    voteScore={0}
                  />
</div>,
                <div
                  className="category"
>
                  [
                  
                  ]
</div>,
              ],
              "className": "showpostnav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Votes
                    onDownVote={[Function]}
                    onUpVote={[Function]}
                    voteScore={0}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "onDownVote": [Function],
                    "onUpVote": [Function],
                    "voteScore": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "[",
                    "",
                    "]",
                  ],
                  "className": "category",
                },
                "ref": null,
                "rendered": Array [
                  "[",
                  "",
                  "]",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
